conans/client/profile_loader.py

```diff
@@ -137,7 +137,7 @@ def _load_profile(text, profile_path, default_folder):
         for include in profile_parser.get_includes():
             # Recursion !!
             profile, included_vars = read_profile(include, cwd, default_folder)
-            inherited_profile.update(profile)
+            inherited_profile.update_profile(profile)
```

Comment:
IMO `profile.update_profile` looks too redundand. `profile.update` is nice and self-explaining.

Reply:
Until you try to search for occurrences, or code calling this method and it becomes almost impossible to know, and you need to rely on CI running to find all occurrences.

Reply:
Agree that the ``update_profile()`` is not great and redundant. I have replaced it for ``compose()`` because it doesn't conflict with common dicts update() and also because in our docs we refer to this operation as "Profile composition".